op-node/rollup/derive/engine_queue.go

```diff
@@ -390,7 +420,7 @@ func (eq *EngineQueue) logSyncProgress(reason string) {
 // this is a no-op if the nodes already agree on the forkchoice state.
 func (eq *EngineQueue) tryUpdateEngine(ctx context.Context) error {
 	fc := eth.ForkchoiceState{
-		HeadBlockHash:      eq.unsafeHead.Hash,
+		HeadBlockHash:      eq.engineSyncTarget.Hash,
```

Comment:
When this code is called, it is always `unsafeHead == engineSyncTarget`.
Lets check this invariant is satisfied. If not, lets log for this incorrect behavior